lucene/misc/src/test/org/apache/lucene/misc/store/TestDirectIODirectory.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.misc.store;
+
+import com.carrotsearch.randomizedtesting.LifecycleScope;
+import com.carrotsearch.randomizedtesting.RandomizedTest;
+import org.apache.lucene.store.*;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.apache.lucene.misc.store.DirectIODirectory.DEFAULT_MIN_BYTES_DIRECT;
+
+public class TestDirectIODirectory extends BaseDirectoryTestCase {
+  public void testWriteReadWithDirectIO() throws IOException {
+    try(Directory dir = getDirectory(RandomizedTest.newTempDir(LifecycleScope.TEST))) {
+      final long blockSize = Files.getFileStore(createTempFile()).getBlockSize();
+      final long minBytesDirect = Double.valueOf(Math.ceil(DEFAULT_MIN_BYTES_DIRECT / blockSize)).longValue() *
+                                    blockSize;
+      // Need to worry about overflows here?
+      final int writtenByteLength = Math.toIntExact(minBytesDirect);
+
+      MergeInfo mergeInfo = new MergeInfo(1000, Integer.MAX_VALUE, true, 1);
+      final IOContext context = new IOContext(mergeInfo);
+
+      IndexOutput indexOutput = dir.createOutput("test", context);
+      indexOutput.writeBytes(new byte[writtenByteLength], 0, writtenByteLength);
+      IndexInput indexInput = dir.openInput("test", context);
+
+      assertEquals("The length of bytes read should equal to written", writtenByteLength, indexInput.length());
+
+      indexOutput.close();
+      indexInput.close();
+    }
+  }
+
+  @Override
+  protected Directory getDirectory(Path path) throws IOException {
```

Comment:
Thanks great. We have to check, if the base class also sometimes send the correct IOContexts, so our directory is triggered. Because the wrong IOContext will cause everything to be delegated to the underlying FSDirectory.

Reply:
I verified: `BaseDirectoryTestCase` uses [newIOContext(Random)](https://lucene.apache.org/core/8_7_0/test-framework/org/apache/lucene/util/LuceneTestCase.html#newIOContext-java.util.Random-). So we have the correct randomization and our directory is used from time to time.

Reply:
Cool thanks for the verification!